### DISCUSSION TITLE
Is basecode treated like a submission?

### DISCUSSION THREAD

**Discussion by csidirop** [jplag/JPlag#261] [Q&A] [answered]
It looks like basecode is treated like a submission:
```
java -jar .\jplag-3.0.0-SNAPSHOT-jar-with-dependencies.jar -l java9 .\SUB\ -bc .\BASECODE\
Initialized language Javac 1.9+ based AST plugin
Basecode directory ".\SUB\\.\BASECODE\" will be used
JPlag initialized
------ Parsing submission: BASECODE
OK
------ Parsing submission: sub1
OK
------ Parsing submission: sub2
OK
------ Parsing submission: sub3
OK
------ Parsing submission: sub4
OK
------ Parsing submission: sub5
OK

6 submissions parsed successfully!
0 parser errors!



Total time for parsing: 0 sec
Time per parsed submission: 45 msec

Comparing BASECODE-sub1: 33.624454
Comparing BASECODE-sub2: 18.81919
Comparing BASECODE-sub3: 29.872496
Comparing BASECODE-sub4: 20.872866
Comparing BASECODE-sub5: 21.761658
Comparing sub1-sub2: 51.47453
Comparing sub1-sub3: 52.58964
Comparing sub1-sub4: 57.45554
Comparing sub1-sub5: 43.67816
Comparing sub2-sub3: 64.994026
Comparing sub2-sub4: 75.582825
Comparing sub2-sub5: 65.0519
Comparing sub3-sub4: 68.61314
Comparing sub3-sub5: 62.70023
Comparing sub4-sub5: 63.14554
Total time for comparing submissions: 0 sec
Writing report...
```

Do I see that correctly? If yes that this redundant.

Tested on Windows 10 with jplag build from source (because the last official version seams to be long outdated).

Edit: pasted the whole output and cl input

**Comment by tsaglam** [comment, marked as answer]
In short: It is not really treating it as a submission, only internally. If you check the generated report, you will see that only the real submissions are listed. 

Internally, the basecode is a special submission and is used for the comparison of normal submission.
In order to do that, each submission is compared to the basecode. Then, when comparing two normal submissions, the basecode match of both submissions is factored into the comparison (meaning, matching sections between the submissions are ignored if they are also matching with the basecode). The reason why the CLI prints the basecode comparisons is so the user can monitor the progress of the comparison process.

See this example report for two submissions and a basecode:
![grafik](https://user-images.githubusercontent.com/4396919/138826809-fcda9fbe-2e75-4bd3-83d9-20d859461920.png)

**Comment by csidirop** [comment]
Okay that makes sence, thanks!
